components/Reader/Reader.js

```diff
@@ -8,71 +8,234 @@
 import { Router } from '../../routes';
 
-type Props = {
-  book: Book,
-  onClose(): void,
-  chapter: number,
-};
-
-type State = {
-  chapters: { [number]: Chapter },
-};
+/* eslint-disable react/no-multi-comp */
 
 function createMarkup(chapter: Chapter) {
   return { __html: chapter.content };
 }
 
 const Card = styled.div`
-  ${media.tablet`border-radius: 4px;`} background: ${props =>
-      props.theme.grays.white};
-  position: relative;
+  background: ${props => props.theme.grays.white};
+  min-height: 100vh;
   width: 100%;
-  height: 100%;
   display: flex;
   flex-direction: column;
 `;
 
-class Reader extends React.Component<Props, State> {
-  static defaultProps = {
-    chapter: 1,
-  };
+type ReaderProps = {
+  book: Book,
+  onRequestClose(): void,
+  chapter: ?Chapter,
+  chapterNumber: number,
+  onRequestNext(): void,
+  onRequestPrevious(): void,
+};
+
+type ReaderState = {
+  showOverlay: boolean,
+};
 
+const OVERLAY_TIMEOUT = 3000; // 3 seconds
+
+class Reader extends React.PureComponent<ReaderProps, ReaderState> {
   state = {
-    chapters: {},
+    showOverlay: false,
   };
 
-  componentDidMount() {
-    this.loadChapter(this.props.chapter);
+  componentWillUnmount() {
+    // Make sure we clean up after ourselves so we don't try to setState after we've unmounted
+    if (this.timerId) {
+      window.clearTimeout(this.timerId);
+    }
+  }
+
+  onTap = (event: SyntheticTouchEvent<>) => {
+    // If we clicked a button, ignore the tap
+    if (event.target.type === 'button') {
+      return;
+    }
+
+    // Toggle the overlay and clear/set timer accordingly
+    this.setState(
+      state => ({ showOverlay: !state.showOverlay }),
+      () => {
+        if (this.state.showOverlay) {
+          this.timerId = window.setTimeout(
+            () => this.setState({ showOverlay: false }),
+            OVERLAY_TIMEOUT,
+          );
+        } else {
+          window.clearTimeout(this.timerId);
+        }
+      },
+    );
+  };
+
+  timerId: number;
+
+  render() {
+    const { book, chapter, chapterNumber } = this.props;
+    const numOfChapters = book.chapters.length;
+
+    const disableNext = chapterNumber >= numOfChapters;
+    const disablePrev = chapterNumber <= 1;
+
+    return (
+      <Container px={0} mw={1075}>
+        <Backdrop />
+        <Swipeable
+          onSwipedLeft={this.props.onRequestNext}
+          onSwipedRight={this.props.onRequestPrevious}
+          onTap={this.onTap}
+        >
+          <ButtonOverlay
+            showOnMobile={this.state.showOverlay}
+            onRequestNext={this.props.onRequestNext}
+            onRequestPrev={this.props.onRequestPrevious}
+            disableNext={disableNext}
+            disablePrev={disablePrev}
+          />
+          <Card>
+            <Toolbar
+              showOnMobile={this.state.showOverlay}
+              onRequestClose={this.props.onRequestClose}
+            />
+            <Box px={[40, 120]} pb={20} flex="1 0 auto">
```

Comment:
Mer et spørsmål enn en kommentar. Klarer man å holde oversikt over alle slike verdier, når de er hardkodet rundt omkring in applikasjonen?

Reply:
På sikt ser jeg for oss at vi får til en slags spacing scale